tests/test_sft_trainer.py

```diff
@@ -808,3 +809,62 @@ def test_pretokenized_dataset_wrong_format():
         # is essentially swallowing a KeyError here.
         with pytest.raises(ValueError):
             sft_trainer.train(MODEL_ARGS, data_args, train_args, PEFT_PT_ARGS)
+
+
+def test_set_log_level_for_logger_default():
+    """
+    Ensure that the correct log level is being set
+    for python native logger and transformers logger
+    """
+
+    # Set env var TRANSFORMERS_VERBOSITY as None and test
+    os.unsetenv("TRANSFORMERS_VERBOSITY")
+    os.unsetenv("LOG_LEVEL")
+    os.unsetenv("SFT_TRAINER_CONFIG_JSON_ENV_VAR")
+    os.unsetenv("SFT_TRAINER_CONFIG_JSON_PATH")
+    train_args = copy.deepcopy(TRAIN_ARGS)
+
+    # TEST IF NO ENV VAR ARE SET AND NO CLI ARGUMENT IS PASSED
+    training_args, logger = sft_trainer.set_log_level(train_args)
+    assert logger.getEffectiveLevel() == logging.WARNING
+    assert training_args.log_level in ["passive", "warning"]
+
+
+def test_set_log_level_for_logger_with_env_var():
+    """
+    Ensure that the correct log level is being set
+    for python native logger and transformers logger
+    """
+
+    os.unsetenv("TRANSFORMERS_VERBOSITY")
+    os.unsetenv("LOG_LEVEL")
+    os.unsetenv("SFT_TRAINER_CONFIG_JSON_ENV_VAR")
+    os.unsetenv("SFT_TRAINER_CONFIG_JSON_PATH")
+    train_args_env = copy.deepcopy(TRAIN_ARGS)
+
+    # TEST IF LOG_LEVEL ENV VAR IS SET AND NO CLI ARGUMENT IS PASSED
+    os.environ["LOG_LEVEL"] = "info"
```

Comment:
Similarly, an easier way of doing this would be to set:
```py
# decorator can be used on whole function
@mock.patch.dict(os.environ, {"LOG_LEVEL": "info"}, clear=True)

# or can create context scope
with mock.patch.dict(os.environ, {"LOG_LEVEL": "info"}, clear=True):
```